update README, improve markdown formatting, specify `long_description_content_type` to markdown, add versioning and GH Action for PyPI release (#24)
datajoint/element-array-ephys: README.md

    + An electrode here refers to one recordable electrode site on the Neuropixels probe (e.g. for Neuropixels 1.0, there are 960 sites per shank)
+ ***Probe*** - record of an actual physical probe, identifiable by some unique ID (e.g. probe's serial number)
+ ***ElectrodeConfig*** - particular electrode configuration to be used for ephys recording
+ ***ElectrodeConfig.Electrode*** - corresponding electrodes in ***ProbeType.Electrode*** that are used for recording in this electrode configuration (e.g. for Neuropixels 1.0 or 2.0, there can be at most 384 electrodes usable for recording per probe)

### Extracellular ephys recording

> + ***ProbeInsertion*** - a surgical insertion of a probe onto the animal. Every experimental session consists of one or more ***ProbeInsertion***, with corresponding ***InsertionLocation***

Reviewer: ```suggestion
+ ***ProbeInsertion*** - a surgical insertion of a probe in the brain. Every experimental session consists of one or more entries in ***ProbeInsertion*** with a corresponding ***InsertionLocation*** each
```

+ ***EphysRecording*** - each ***ProbeInsertion*** is accompanied by a corresponding ***EphysRecording***, specifying the ***ElectrodeConfig*** used for the recording from the ***Probe*** defined in such ***ProbeInsertion***
    
### Clusters and spikes

This ephys element features automatic ingestion for spike sorting results from the ***kilosort*** method. 

+ ***Clustering*** - specify instance(s) of clustering on an ***EphysRecording***, by some ***ClusteringMethod***
+ ***Curation*** - specify instance(s) of curations performed on the output of a given ***Clustering***
+ ***CuratedClustering*** - set of results from a particular round of clustering/curation
    + ***CuratedClustering.Unit*** - Identified unit(s) from one ***Curation***, and the associated properties (e.g. cluster quality, spike times, spike depths, etc.)

>     + ***WaveformSet*** - A set of spike waveforms for units out of a given CuratedClustering

Reviewer: ```suggestion
    + ***WaveformSet*** - A set of spike waveforms for units from a given CuratedClustering
```
